From 85fc7cbf831e914456623a2f7a9efbe81919bf4d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Olivier=20Cr=C3=AAte?= <olivier.crete@collabora.com>
Date: Tue, 28 Apr 2015 19:46:54 -0400
Subject: local-metadata: Ignore sanitization if the whole filename is
 blacklisted

If the whole filename ends up blacklisted, then just take it as-is.

https://bugzilla.gnome.org/show_bug.cgi?id=748604

diff --git a/src/local-metadata/grl-local-metadata.c b/src/local-metadata/grl-local-metadata.c
index adc81f0..45856be 100644
--- a/src/local-metadata/grl-local-metadata.c
+++ b/src/local-metadata/grl-local-metadata.c
@@ -272,6 +272,13 @@ video_sanitise_string (const gchar *str)
   if (*line_end != '\0') {
     line_end = g_utf8_find_prev_char (line, line_end);
 
+
+    /* If everything in the string is blacklisted, just ignore
+     * the blackisting logic.
+     */
+    if (line_end == NULL)
+      return g_strdup (str);
+
     /* After removing substring with blacklisted word, ignore non alpha-numeric
      * char in the end of the sanitised string */
     while (g_unichar_isalnum (*line_end) == FALSE &&
-- 
cgit v0.10.2

